app/controllers/error.js

```diff
@@ -0,0 +1,12 @@
+import Controller from '@ember/controller';
+import { NotFoundError } from '@ember-data/adapter/error';
+
+export default class ErrorController extends Controller {
+  get error() {
+    return this.model;
+  }
+
+  get isNotFoundError() {
+    return this.error instanceof NotFoundError;
+  }
```

Comment:
We could also do this check in the `error` action in the application route, and redirect to the `route-not-found` page instead, but that way we also need to construct the URL, so this solution is simpler.